### PR TITLE
[IMP] payment, *: verify and reroute payment flows

### PR DESCRIPTION
**Pull request by anko-odoo**
This commit changes the way transactions linked to a document (sales
order, invoice...) are created in a payment flow. Rather than receiving
and trusting the transaction values from the controller, they are now
read from the linked document, and the payment flow is rerouted to use
the document's module's controllers instead of that of `payment`.

This ensures that no unexpected value can be passed to the `create`
method of a transaction, and simplifies the implementation of the
payment flows of linked documents.

task-3136240

See also:
- https://github.com/odoo/enterprise/pull/43212
- https://github.com/odoo/upgrade/pull/5124

### CODE REVIEW

**Review by AntoineVDV — APPROVED** [review]
@robodoo r+

**Comment by AntoineVDV** [comment]
@robodoo r+

**Comment by AntoineVDV** [comment]
@robodoo r+

**Comment by AntoineVDV** [comment]
@robodoo r+

File: addons/payment/static/src/js/express_checkout_form.js [hunk 1]
```diff
@@ -58,12 +58,9 @@ publicWidget.registry.PaymentExpressCheckoutForm = publicWidget.Widget.extend({
...
     _prepareTransactionRouteParams(providerId) {
         return {
-            'payment_option_id': parseInt(providerId),
-            'reference_prefix': this.paymentContext['referencePrefix'] &&
-                                this.paymentContext['referencePrefix'].toString(),
-            'currency_id': this.paymentContext['currencyId'] &&
-                           parseInt(this.paymentContext['currencyId']),
-            'partner_id': parseInt(this.paymentContext['partnerId']),
+            'provider_id': parseInt(providerId),
+            'payment_method_id': 1, // TODO VCR
```
Comment (Feyensv) [inline addons/payment/static/src/js/express_checkout_form.js:62]: @chevalierv FYI

**Pull request closed without merging**